vscode-dotnet-runtime-library/src/Utils/WebRequestWorker.ts

```diff
@@ -2,57 +2,164 @@
+        private readonly websiteTimeoutMs: number, // Match the default timeout time of 10 minutes.
+        private cacheTimeToLive = -1
+        )
+        {
+            this.cacheTimeToLive = this.cacheTimeToLive === -1 ? this.websiteTimeoutMs * 5 : this.cacheTimeToLive; // make things live 5x (arbitrary) as long as it takes to maximally acquire them
+            const uncachedAxiosClient = Axios.create({});
+            Debugging.log(`Axios client instantiated: ${uncachedAxiosClient}`);
 
-    public async getCachedData(retriesCount = 2): Promise<string | undefined> {
-        this.cachedData = this.extensionState.get<string>(this.extensionStateKey);
-        if (!this.cachedData) {
-            // Have to acquire data before continuing
-            this.cachedData = await this.makeWebRequestWithRetries(true, retriesCount);
-        } else if (!this.currentRequest) {
-            // Update without blocking, continue with cached information
-            this.currentRequest = this.makeWebRequest(false);
-            this.currentRequest.then((result) => {
-                if (result) {
-                    this.cachedData = result;
+            // Wrap the client with a retry interceptor. We don't need to return a new client, it should be applied automatically.
+            axiosRetry(uncachedAxiosClient, {
+                // Inject a custom retry delay to expoentially increase the time until we retry.
+                retryDelay(retryCount: number) {
+                    return Math.pow(2, retryCount); // Takes in the int as (ms) to delay.
                 }
-                this.currentRequest = undefined;
             });
+
+            Debugging.log(`Axios client wrapped around axios-retry: ${uncachedAxiosClient}`);
+
+            this.client = setupCache(uncachedAxiosClient,
+                {
+                    storage: mementoStorage(this.extensionState),
+                    ttl: this.cacheTimeToLive
+                }
+            );
+
+            Debugging.log(`Cached Axios Client Created: ${this.client}`);
+    }
+
+    /**
+     *
+     * @param url The URL of the website to send a get request to.
+     * @param options The AXIOS flavor options dictonary which will be forwarded to an axios call.
+     * @returns The response from AXIOS. The response may be in ANY type, string by default, but maybe even JSON ...
+     * depending on whatever the request return content can be casted to.
+     * @remarks This function is used as a custom axios.get with a timeout because axios does not correctly handle CONNECTION-based timeouts:
+     * https://github.com/axios/axios/issues/647 (e.g. bad URL/site down).
+     */
+    private async axiosGet(url : string, options = {})
+    {
+        if(url === '' || !url)
+        {
+            throw new Error(`Request to the url ${this.url} failed, as the URL is invalid.`);
         }
-        return this.cachedData;
+        const abort = Axios.CancelToken.source()
```

Comment:
This doesnt seem to actually set a timeout like the function suggests?

Reply:
its private and all of the callers do, but according to the comment above, axios timeout does not work, and I agree with those comments. Dont know why axios has not done anything about it

Reply:
Oh yes its the connection based vs response based timeout fiasco, yes I should ideally add a timeout, will cc offline to see if this is a release blocker

Reply:
I added a custom timeout which seems to work.